
/*
      Dopo aver modificato il .proto file:
          python3 -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/gp_manager.proto

      Questo genera
          - gp_manager_pb2.py che contiene le classi di richiesta e risposta
          - gp_manager_pb2_grpc.py che contiene le classi client e server

*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "/SDCC/services";
option java_outer_classname = "GpManagerProto";
option objc_class_prefix = "GPM";
package gp_manager;

service GpCreator {
  rpc groupCreate (gpCreateReq) returns (gpCreateRep) {}
}

message gpCreateReq {
  string group_name = 1;
  string username = 2;
  string email = 3;
  string service = 4;
}

message gpCreateRep {
  bool isCreated = 1;
}